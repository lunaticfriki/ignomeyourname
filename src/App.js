import React, { useContext } from 'react'
import { Router } from '@reach/router'
import { GnomeContext } from './context/gnomeContext'

import Layout from './components/layout'
import Main from './components/main'
import About from './components/about'

import Gallery from './components/gallery'

function App() {
  const { gnomes, gnome, setGnome } = useContext(GnomeContext)
  return (
    <div className="App">
      <Layout>
        <Router>
          <Main path="/" gnome={gnome} gnomes={gnomes} setGnome={setGnome} />
          <About path="/about" />
          <Gallery path="/gallery" gnome={gnome} gnomes={gnomes} setGnome={setGnome} />
        </Router>
      </Layout>
    </div>
  )
}

export default App
