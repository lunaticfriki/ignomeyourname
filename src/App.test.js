import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import App from './App'
import store from './store/index'

describe('App component suite', () => {
  test('App snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
