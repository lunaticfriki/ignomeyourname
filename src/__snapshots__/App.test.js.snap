// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component suite App snapshot 1`] = `
<div
  className="sc-AxjAm lauOEi App"
>
  <div
    className="sc-AxhCb iCgYvc"
  >
    <div
      className="sc-AxirZ cToqvt"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#d62b9a",
          }
        }
      >
        <h1>
          I GNOME YOUR NAME!
        </h1>
      </a>
      <h2>
        Search your favorite gnome from Brastlewark!
      </h2>
      <nav>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#d62b9a",
            }
          }
        >
          <li>
            Searcher
          </li>
        </a>
        <a
          href="/gallery"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          <li>
            Gallery
          </li>
        </a>
        <a
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          <li>
            About me
          </li>
        </a>
      </nav>
    </div>
    <div
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="-1"
    >
      <div
        className="sc-AxhUy cuGvRN"
      >
        <div
          className="sc-AxgMl cgccNm"
        >
          <form
            onSubmit={[Function]}
          >
            <label
              className="c-use-gnome"
              htmlFor="use-gnome-findyour gnome"
            >
              Find your gnome
              <select
                data-testid="use-gnome-findyour gnome"
                disabled={true}
                id="use-gnome-findyour gnome"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
              >
                <option
                  hidden={true}
                >
                   Choose your gnome!
                </option>
              </select>
            </label>
            <button>
              Search!
            </button>
          </form>
        </div>
        <div
          className="sc-AxheI cpImOa"
        />
      </div>
    </div>
    <div
      className="sc-AxiKw llWKsH"
    >
      <i>
        oscar ventura - gnome searcher - 2020
      </i>
    </div>
  </div>
</div>
`;
